codersdk/deployment.go

```diff
@@ -686,11 +686,15 @@ type NotificationsConfig struct {
 	Webhook NotificationsWebhookConfig `json:"webhook" typescript:",notnull"`
 }
 
+func (n *NotificationsConfig) Enabled() bool {
+	return n.SMTP.Smarthost != "" || n.Webhook.Endpoint != serpent.URL{}
```

Comment:
We already have a validation function for smarthost; let's centralise that logic (i.e. check for `ValidationNoSmarthostErr`).

We could do the same for webhook. If both return errors, neither are configured correctly.